daemon/mgr/cri_network.go

```diff
@@ -0,0 +1,115 @@
+package mgr
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/cri-o/ocicni/pkg/ocicni"
+	"github.com/sirupsen/logrus"
+
+	"github.com/alibaba/pouch/daemon/config"
+
+	"k8s.io/kubernetes/pkg/kubelet/apis/cri/v1alpha1/runtime"
+)
+
+// CniMgr as an interface defines all operations against CNI.
+type CniMgr interface {
+	// Name returns the plugin's name. This will be used when searching
+	// for a plugin by name, e.g.
+	Name() string
+
+	// SetUpPod is the method called after the sandbox container of the
+	// pod has been created but before the other containers of the pod
+	// are launched.
+	SetUpPod(config *runtime.PodSandboxConfig, id string, netnsPath string) error
+
+	// TearDownPod is the method called before a pod's sandbox container will be deleted.
+	TearDownPod(config *runtime.PodSandboxConfig) error
+
+	// GetPodNetworkStatus is the method called to obtain the ipv4 or ipv6 addresses of the pod sandbox.
+	GetPodNetworkStatus(config *runtime.PodSandboxConfig) (string, error)
+
+	// Status returns error if the network plugin is in error state.
+	Status() error
+}
+
+// CniManager is an implementation of interface CniMgr.
+type CniManager struct {
+	// plugin is used to setup and teardown network when run/stop pod sandbox.
+	plugin ocicni.CNIPlugin
+}
+
+// NewCniManager initializes a brand new cni manager.
+func NewCniManager(cfg *config.Config) (*CniManager, error) {
```

Comment:
If you just use `cfg.CriConfig`, we can pass a parameter like `cfg.CriConfig` rather than `cfg config.Config`? WDYT?